### PR TITLE
EIP 2929 - dynamic state access costs, by cache warmth

### LINKED ISSUE #1974: State access gas cost changes: EIP-2929 for Berlin
Implement https://eips.ethereum.org/EIPS/eip-2929 for Berlin

### LINKED ISSUE #1975: State access lists in transactions: EIP-2930 for Berlin
Implement https://eips.ethereum.org/EIPS/eip-2930 for Berlin

Depends on #1973 and #1974

### PR DESCRIPTION
**Pull request by carver**
### What was wrong?

Fix #1974 
Fix #1975 

### How was it fixed?

Replace a bunch of opcodes that dynamically check the cache warmth and charge appropriately.

Bonus changes:
- show stack in debug2 traces
- refactor opcode tests to use less custom hackery and more actual computation-execution code (necessary for the `BerlinTransactionExecutor` implementation to be testable)
- renamed the variables in the sstore logic to match the EIP-2200 names

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history
- [x] Pass all test cases from this EIP-referenced test case: https://gist.github.com/holiman/174548cad102096858583c6fbbb0649a
  - [x] 1
  - [x] 2
  - [x] 3
  - [x] 4
- [x] Add entry to the [release notes](https://github.com/ethereum/py-evm/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by carver** [comment]
Fun, the new sstore costs make the benchmarking fail to fit in the block :+1: :+1:

**Comment by carver** [comment]
@kclowes So I'm still pretty sure that the revert logic is not correct, but I think this is far enough along to start getting an early look. I'm happy to do a walkthrough tomorrow if you're around.

**Comment by kclowes** [comment]
@carver awesome! Happy to do a walkthrough any time!

File: eth/db/account.py [hunk 1]
```diff
@@ -391,6 +425,7 @@ def commit(self, checkpoint: JournalDBCheckpoint) -> None:
     def lock_changes(self) -> None:
```
Comment (carver) [inline eth/db/account.py:425]: Maybe call `end_previous_txn` or something better.

File: eth/db/account.py [hunk 2]
```diff
@@ -174,6 +181,22 @@ def delete_storage(self, address: Address) -> None:
...
         self._wipe_storage(address)
 
+    def is_storage_warm(self, address: Address, slot: int) -> bool:
+        key = self._get_storage_tracker_key(address, slot)
+        return key in self._journal_accessed_storage
+
+    def mark_storage_warm(self, address: Address, slot: int) -> None:
+        key = self._get_storage_tracker_key(address, slot)
+        if key not in self._journal_accessed_storage:
+            self._journal_accessed_storage[key] = IS_PRESENT_VALUE
```
Comment (carver) [inline eth/db/account.py:191]: Combine db with account access list. Keys are always disjoint

File: eth/db/account.py (hunk 1 above)
Comment (carver) [reply eth/db/account.py:431]: Moved to #1990

**Comment by carver** [comment]
Verbal :+1: from @kclowes

**Pull request merged**